pkg/controllers/dynakube/version/oneAgentVersionReconciler.go

```diff
@@ -0,0 +1,212 @@
+		return false
+	}
+
+	if current.Source != previous.Source {
+		log.Info("source changed, update for version status is needed", "updater", "OneAgent")
+		return true
+	}
+	if previous.CustomImageNeedsReconciliation(logCustomOneAgentImageChangedMessage(current, previous), dynakube.CustomOneAgentImage()) {
+		return true
+	}
+	if previous.CustomVersionNeedsReconciliation(logCustomVersionChangedMessage(current, previous), dynakube.CustomOneAgentVersion()) {
+		return true
+	}
+
+	return isOutdated(logSkipUpdateTimestampValidMessage("OneAgent"), reconciler.timeProvider, dynakube, dynakube.Status.OneAgent.LastProbeTimestamp)
+}
+
+// log function to print messages
+func logCustomOneAgentImageChangedMessage(current status.VersionStatus, previous status.VersionStatus) func() {
+	return func() {
+		log.Info("custom image value changed, update for version status is needed", "updater", "OneAgent", "oldImage", previous.ImageID, "newImage", current.ImageID)
+	}
+}
+
+// another log function to print messages
+func logCustomVersionChangedMessage(current status.VersionStatus, previous status.VersionStatus) func() {
+	return func() {
+		log.Info("custom version value changed, update for version status is needed", "updater", "OneAgent", "oldVersion", previous.Version, "newVersion", current.Version)
+	}
+}
+
+func (reconciler *OneAgentImageVersionReconciler) run(ctx context.Context, current status.VersionStatus, dynakube *dynatracev1beta1.DynaKube) (status.VersionStatus, error) {
+	previous := dynakube.Status.OneAgent.VersionStatus
+	if current.Source == status.CustomImageVersionSource {
+		log.Info("updating version status according to custom image", "updater", "OneAgent")
+
+		return updateVersionWithImage(ctx, reconciler.registryClient, current, dynakube.CustomOneAgentImage())
+	}
+
+	if !dynakube.ShouldAutoUpdateOneAgent() {
+		if InitialUpdateInProgress(previous) {
+			log.Info("initial status update in progress with no auto update", "updater", "OneAgent")
+		} else if current.Source == previous.Source {
+			log.Info("status updated skipped, due to no auto update", "updater", "OneAgent")
+			return previous, nil
+		}
+	}
+
+	if current.Source == status.PublicRegistryVersionSource {
+		return reconciler.handlePublicRegistry(ctx, current, dynakube, previous)
+	}
+	return reconciler.handleTenantRegistry(ctx, current, dynakube, previous)
+}
+
+func (reconciler *OneAgentImageVersionReconciler) handleTenantRegistry(ctx context.Context, current status.VersionStatus, dynakube *dynatracev1beta1.DynaKube, previous status.VersionStatus) (status.VersionStatus, error) {
+	latestVersion := dynakube.CustomOneAgentVersion()
+	var err error
+	if latestVersion == "" {
+		latestVersion, err = reconciler.dtClient.GetLatestAgentVersion(dtclient.OsUnix, dtclient.InstallerTypeDefault)
+		if err != nil {
+			return previous, err
+		}
+	}
+
+	downgrade, err := checkForDowngrade(previous, latestVersion)
```

Comment:
can we name it in same manner? E.g. `previous`and `latest` or with `Version` suffix both variables?